integrated_channels/integrated_channel/management/commands/transmit_courseware_data.py

```diff
@@ -0,0 +1,133 @@
+    SAPSuccessFactorsEnterpriseCustomerConfiguration: SapCourseExporter,
+}
+
+LOGGER = getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Transmit courseware data to the IntegratedChannel(s) linked to any or all EnterpriseCustomers.
+    """
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            '--enterprise_customer',
+            dest='enterprise_customer',
+            default=None,
+            metavar='ENTERPRISE_CUSTOMER_UUID',
+            help='Transmit data for only this EnterpriseCustomer. '
+            'Omit this option to transmit to all EnterpriseCustomers with active integrated channels.'
+        )
+        parser.add_argument(
+            '--catalog_user',
+            dest='catalog_user',
+            default=None,
+            metavar='ENTERPRISE_CATALOG_API_USERNAME',
+            help='Use this user to access the Course Catalog API. Omit this option to use the'
+            ' username specified in the EDXAPP_ENTERPRISE_CATALOG_API_USERNAME setting.'
+        )
+
+    def handle(self, *args, **options):
+        """
+        Transmit the courseware data for the EnterpriseCustomer(s) to the active integration channels.
+        """
+        enterprise_customers = []
+        # If a valid enterprise customer uuid was provided, transmit for that customer.
+        # Otherwise, transmit for all customers.
+        enterprise_customer_uuid = options['enterprise_customer']
+        if enterprise_customer_uuid:
+            try:
+                enterprise_customer = EnterpriseCustomer.active_customers.get(uuid=enterprise_customer_uuid)
+            except EnterpriseCustomer.DoesNotExist:
+                raise CommandError('Enterprise customer {} not found, or not active'.format(enterprise_customer_uuid))
+            else:
+                enterprise_customers.append(enterprise_customer)
+        else:
+            enterprise_customer = None
+
+        # Find the correct user with which to access the catalog API
+        manual_catalog_username = options['catalog_user']
+        settings_catalog_username = getattr(settings, 'EDXAPP_ENTERPRISE_CATALOG_API_USERNAME', None)
+        username = manual_catalog_username or settings_catalog_username
+
+        user = self.validate_user(username)
+
+        enterprise_customers = enterprise_customers or EnterpriseCustomer.active_customers.all()
+
+        for customer in enterprise_customers:
+            try:
+                # We're running with Celery, and want to send the task to be processed asynchronously
+                send_data_task.delay(user.username, customer.uuid)
+            except AttributeError:
+                # Celery isn't available; run the task inline
+                send_data_task(user.username, customer.uuid)
+
+
+    @staticmethod
+    def validate_user(username):
+        if not username:
+            raise CommandError('A username must be specified manually or in settings to access the course catalog API.')
```

Comment:
So, with the above (removing the ability to set a global worker username), we can just flag this arg as required. Problem solved!